### PR TITLE
Update DB schema and  GraphQL APIs for 3rd party app

### PR DESCRIPTION
**Pull request by Junoth**
All the update follows design doc except default value of active_version_number. I think we can follow the original design to mark that field as NULL if there is no active version instead of setting 0 as default value.

https://docs.google.com/document/d/1LPBU4WjMJrpX075Bp4yJqQSjyY75x1VOWXfE2oYtEnI/edit?usp=sharing

### CODE REVIEW

File: core/api/gql/schema.graphql [hunk 1]
```diff
@@ -280,19 +285,20 @@ type App {
 type AppVersion {
 	app: App!
 	versionNumber: Int!
-	appName: String!
 	iconUrl: String
 	hasUiExtension: Boolean!
 	uiExtensionEntrypointPath: String
 	isPublic: Boolean!
 	visibleToTeams: [Team!]!
+	changeLog: String!
```
Comment (naturetime10) [inline core/api/gql/schema.graphql]: Can we rename this to `changes`? `Changelog` contains a curated, chronologically ordered list of notable changes for `each version` of a project.

File: core/api/gql/schema.graphql [hunk 2]
```diff
@@ -401,13 +408,19 @@ input FileMetadataInput {
...
+	description: String
 	activeVersionNumber: Int
 }
 
 input UpdateAppVersionInput {
-	name: String!
 	iconUrl: String
 	hasUiExtension: Boolean!
 	uiExtensionEntryPointPath: String
+	changeLog: String
```
Comment (naturetime10) [inline core/api/gql/schema.graphql]: Can we rename this to `changes`? `Changelog` contains a curated, chronologically ordered list of notable changes for `each version` of a project.

File: migrations/20230130200400_update_app_tables.sql [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	DROP COLUMN description,
+	DROP COLUMN app_name;
+
+ALTER TABLE app_version
+	DROP COLUMN change_log,
+    ADD COLUMN app_name VARCHAR(100) NOT NULL;
+
+ALTER TABLE app_installation
+	DROP COLUMN enabled_version_number;
+
```
Comment (naturetime10) [inline migrations/20230130200400_update_app_tables.sql]: Let's do exact reverse of migration up to ensure we can also undo the changes correct.

File: migrations/20230130200400_update_app_tables.sql [hunk 4]
```diff
@@ -0,0 +1,24 @@
+-- +migrate Up
+ALTER TABLE app
+	ADD description TEXT,
+    ADD app_name VARCHAR(100) NOT NULL DEFAULT '';
+
+ALTER TABLE app_version
+    ADD change_log TEXT,
```
Comment (naturetime10) [inline migrations/20230130200400_update_app_tables.sql]:
```suggestion
    ADD changes TEXT,
```

File: migrations/20230130200400_update_app_tables.sql [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ALTER TABLE app_installation
+    ADD enabled_version_number INT NOT NULL DEFAULT 0;
+
+-- +migrate Down
+ALTER TABLE app
+	DROP COLUMN description,
+	DROP COLUMN app_name;
+
+ALTER TABLE app_version
+	DROP COLUMN change_log,
```
Comment (naturetime10) [inline migrations/20230130200400_update_app_tables.sql]:
```suggestion
	DROP COLUMN changes,
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: migrations/20230130200400_update_app_tables.sql (hunk 3 above)
Comment (naturetime10) [inline migrations/20230130200400_update_app_tables.sql:35]:
```suggestion
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

File: core/api/gql/schema.graphql (hunk 1 above)
Comment (Junoth) [reply core/api/gql/schema.graphql]: sure. I'll update this

**Review by naturetime10 — APPROVED** [review]

File: core/api/gql/team.go [hunk 6]
```diff
@@ -129,7 +129,7 @@ func (t Team) Sprints(ct context.Context, args struct {
 	}), nil
 }
 
-func (t Team) AppInstallations() ([]AppInstallation, error) {
+func (t Team) AppTeamInstallations() ([]AppTeamInstallation, error) {
```
Comment (naturetime10) [inline core/api/gql/team.go]:
```suggestion
func (t Team) AppInstallations() ([]AppTeamInstallation, error) {
```

File: core/api/gql/schema.graphql [hunk 7]
```diff
@@ -181,7 +184,7 @@ type Team {
 	tasks(filter: TaskFilter): [Task!]!
 	invitations(filter: InvitationFilter): [Invitation!]!
 	sprints(filter: SprintFilter): [Sprint!]!
-	appInstallations: [AppInstallation!]!
+	appTeamInstallations: [AppTeamInstallation!]!
```
Comment (naturetime10) [inline core/api/gql/schema.graphql]:
```suggestion
	appInstallations: [AppTeamInstallation!]!
```

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**